tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+---
+
+- name: Find required providers
+  set_fact:
+    providers: |
+      {{
+        certificate_requests |
+        map(attribute='provider', default=__certificate_provider_default) |
+        unique |
+        list
+      }}
+
+- name: Ensure certmonger is installed
+  package:
+    name: certmonger
+    state: present
+  when: "'certmonger' in providers"
+
+- name: Ensure certificate requests
+  certificate_request:
+    name: "{{ item.name | default(omit) }}"
+    dns: "{{ item.dns | default(omit) }}"
+    ca: "{{ item.ca | default(omit) }}"
+    provider: "{{ item.provider | default(omit) }}"
+  loop: "{{ certificate_requests }}"
```

Comment:
you should have a `notify: certificate handler to restart services` here

How do you apply the changes?  Do you need to restart the certmonger service to tell it to process the new/changed cert requests?  Is there some other command you use to tell certmonger to process?  Does it matter which provider is used?

That is - you might need a different handler depending on what the provider is - you might not need any handlers at all.

Reply:
I don't actually think we'll need handlers for now. I'll remove the handlers directory for now.

Reply:
Handlers removed here: https://github.com/linux-system-roles/certificate/pull/1/commits/7012d594bd5ccbbcb7b03ce7bf0922f2ac9425cd